models/match.py

```diff
@@ -0,0 +1,50 @@
+"""Module containing the Match class and other related components"""
+
+from __future__ import annotations
+from datetime import datetime
+from enum import Enum
+from typing import Optional
+from dataclasses import dataclass
+
+
+@dataclass
+class Match:
+    """A Premier League match between two teams in a particular season.
+
+    Instance Attributes:
+        - teams: A set of the the two teams names that played this match.
+        - date_time: The date and time of the match played.
+        - winner: The name of the winning team of the match, or None if the match ended in a draw.
```

Comment:
Perhaps we should organize the teams and the results by home and away. I'm thinking we say `home` and `away` attribute to refer to each team. Perhaps we should have a `result` attribute that evaluates to a named constant of`HOME_WIN` or `AWAY_WIN` or `DRAW`. Thoughts?

Reply:
I think that would be helpful if we analysed a teams home and away performance separately. We aren't doing that right now so I don't think this would be helpful. Although analysing home and away performance separately is something we could add in our statistics.

Reply:
I think it may be expected that we have home vs away results and that we analyze how that affects performance so maybe we should have it?

Reply:
Yeah home/away team stats woud be nice. I'll make the changes.